aspnetcore/grpc/diagnostics.md

```diff
@@ -99,6 +107,88 @@ dbug: Grpc.Net.Client.Internal.GrpcCall[4]
       Finished gRPC call.
 ```
 
+## Tracing
+
+gRPC services and the gRPC client provide information about gRPC calls using [DiagnosticSource](https://docs.microsoft.com/dotnet/api/system.diagnostics.diagnosticsource) and [Activity](https://docs.microsoft.com/dotnet/api/system.diagnostics.activity). .NET gRPC uses an activity to represent a gRPC call. Tracing events are written to the diagnostic source at the start and stop of the gRPC call activity.
+
+### gRPC service tracing
+
+gRPC services are hosted on ASP.NET Core which reports events about incoming HTTP requests. gRPC specific metadata is added to the existing HTTP request diagnostics that ASP.NET Core provides.
+
+* Diagnostic source name is `Microsoft.AspNetCore`.
+* Activity name is `Microsoft.AspNetCore.Hosting.HttpRequestIn`.
+  * Name of the gRPC method invoked by the gRPC call is added as a tag with the name `grpc.method`.
+  * Status code of the gRPC call when it is complete is added as a tag with the name `grpc.status_code`.
+
+### gRPC client tracing
+
+The .NET gRPC client uses `HttpClient` to make gRPC calls. Although `HttpClient` writes diagnostic events, the .NET gRPC client provides a custom diagnostic source, activity and events so that complete information about a gRPC call can be collected.
+
+* Diagnostic source name is `Grpc.Net.Client`.
+* Activity name is `Grpc.Net.Client.GrpcOut`.
+  * Name of the gRPC method invoked by the gRPC call is added as a tag with the name `grpc.method`.
+  * Status code of the gRPC call when it is complete is added as a tag with the name `grpc.status_code`.
+
+### Collecting tracing
+
+The easist way to use `DiagnosticSource` is to configure a telemetry library such as [Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/asp-net-core) or [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-dotnet) in your app. The library will process information about gRPC calls along-side other app telemetry.
```

Comment:
I believe it worth mentioning that ApplicationInsights and OpenTelemetry do not capture `Grpc.Net.Client.GrpcOut` out of the box for now

Reply:
Good point. The tracing from HttpClient will show up but not gRPC.

I'll add a note. And for the longer term, I'll create issues for AI and OT to add support for GrpcOut + gRPC specific tags.